UIforETW/DirectoryMonitor.cpp

```diff
@@ -62,21 +61,23 @@ DWORD DirectoryMonitor::DirectoryMonitorThread()
 			return 0;
 
 		default:
-			assert(0);
+			ATLASSERT(0);
 			break;
 		}
 	}
   // Unreachable.
 }
 
+_Pre_satisfies_(this->hThread_ == 0)

```

Comment:
Well, _In_ isn't always useless -- if you pass "char*" the it provides some information, although ideally the lack of 'const' would convey that. _InOut_ is more useful. But _In_ is completely meaningless on a type such as DWORD, where it can't even meaningfully/usefully imply _Const_.